src/misc/get-uglify-plugin.js

```diff
@@ -1,5 +1,11 @@
 const webpack = require('webpack');
 let UglifyJsPlugin;
+let isUsingTerser = false;
+
+try {
```

Comment:
```js
try {
  UglifyJsPlugin = require('terser-webpack-plugin');
  isUsingTerser = true;
} catch (e) {}

try {
  UglifyJsPlugin = webpack.optimize.UglifyJsPlugin;
} catch (e) {}

if (!UglifyJsPlugin) {
  try {
    UglifyJsPlugin = require('uglifyjs-webpack-plugin');
  } catch (e) {}
}
```

Regarding this logic, depending on what optimization libs are present, `terser-webpack-plugin` will not be used. For instance, I do not know under what circumstances `webpack.optimize.UglifyJsPlugin` will be available, but given the current implementation, `terser-webpack-plugin` will have the least presidence, is this correct? Maybe explore a different solution, maybe something like:

```js
try {
  UglifyJsPlugin = require('terser-webpack-plugin');
  isUsingTerser = true;

  if (!UglifyJsPlugin) {
    UglifyJsPlugin = webpack.optimize.UglifyJsPlugin;
  }

  if (!UglifyJsPlugin) {
    UglifyJsPlugin = require('uglifyjs-webpack-plugin');
  }
} catch (e) {}
```

?

Alternatively, wrap this in a function with early exits. What do you think?